pyccel/parser/semantic.py

```diff
@@ -1524,91 +1528,112 @@ def _visit_Add(self, expr, **settings):
         elif atoms_str:
             return Concatenate(args, False)
 
-        expr_new = Add(*args, evaluate=False)
+        expr_new = expr.func(*args)
         if stmts:
             expr_new = CodeBlock(stmts + [expr_new])
         return expr_new
 
-    def _visit_Mul(self, expr, **settings):
-        stmts, expr = extract_subexpressions(expr)
-        if stmts:
-            stmts = [self._visit(i, **settings) for i in stmts]
+    def _visit_PyccelMul(self, expr, **settings):
+        #stmts, expr = extract_subexpressions(expr)
+        #if stmts:
+        #    stmts = [self._visit(i, **settings) for i in stmts]
```

Comment:
What is the purpose of leaving these lines commented out instead of removing them?
(The same question applies to several other classes below.)

Reply:
we need to keep them to uncomment them later or move them to another place when we introduce the functional programming

Reply:
Fine with me, but we should make this clear. Maybe add a TODO statement somewhere that we can quickly search?

Reply:
done